src/tools/perf/perftest.c

```diff
@@ -213,6 +214,52 @@ ucs_status_t init_test_params(perftest_params_t *params)
     return UCS_OK;
 }
 
+void free_test_params(perftest_params_t *params)
+{
+    free(params->super.msg_size_list);
+    params->super.msg_size_list = NULL;
+}
+
+static ucs_status_t merge_test_params(struct perftest_context *ctx)
+{
+    const int is_uct          = ctx->params.super.api == UCX_PERF_API_UCT;
+    void *local_rte_group     = ctx->params.super.rte_group;
+    ucx_perf_rte_t *local_rte = ctx->params.super.rte;
+    unsigned needed_flags;
+    char local_dev_name[UCT_DEVICE_NAME_MAX];
+    ucs_status_t status;
+
+    if ((ctx->server_addr != NULL) || ctx->mpi) {
+        return UCS_OK;
+    }
+
+    if (ctx->params.super.api != ctx->peer_params.super.api) {
```

Comment:
seems now mad and socket fail because `ctx->params.super.api` is `UCX_PERF_API_LAST`